lib/convert.ml

```diff
@@ -0,0 +1,388 @@
+open Prelude
+
+module type CONVERT =
```

Comment:
So a design question to consider is: do we want different `Conversion_x` modules in their own separate files?  Probably doesn't matter right now.  But I guess my gut says: once each of them gets big enough, maybe they wanna start going in their own files.  

Anyway, not an actual suggestion just yet, given that we only have one module of signature `CONVERT`, but just to say that if we decide to adopt that design, then I would maybe move the `CONVERT` module sig over to `lib.ml`.

Reply:
Actually that would be a problem, dune will consider that a circularity. If you do want different conversion modules in different files, what I've seen done is to create a separate file, something like `utils.ml`, that just contains all the signature and other globally useful functions.